packages/client/hooks/useIntegrationProviders.tsx

```diff
@@ -0,0 +1,135 @@
+import graphql from 'babel-plugin-relay/macro'
+import {useEffect, useState} from 'react'
+import {IntegrationProviders} from '../types/constEnums'
+import {useIntegrationProvidersGetProvidersListQuery} from '../__generated__/useIntegrationProvidersGetProvidersListQuery.graphql'
+import useAtmosphere from './useAtmosphere'
+
+const query = graphql`
+  query useIntegrationProvidersGetProvidersListQuery($teamId: ID!) {
+    viewer {
+      teamMember(teamId: $teamId) {
+        integrations {
+          atlassian {
+            accessToken
+          }
+          jiraServer {
+            auth {
+              isActive
+            }
+            sharedProviders {
+              id
+            }
+          }
+          github {
+            accessToken
+          }
+          gitlab {
+            auth {
+              provider {
+                scope
+              }
+            }
+            cloudProvider {
+              id
+              clientId
+              serverBaseUrl
+            }
+          }
+          slack {
+            isActive
+          }
+          azureDevOps {
+            auth {
+              accessToken
+            }
+
+            sharedProviders {
+              id
+              tenantId
+              clientId
+            }
+
+            cloudProvider {
+              id
+              tenantId
+              clientId
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+export type Integration = {
+  name: string
+  connected: boolean
+  regex: RegExp
+  provider?: any
+}
+
+const useIntegrationProviders = (teamId: string): Integration[] => {
+  const [providers, setProviders] = useState<Integration[]>([])
+  const atmosphere = useAtmosphere()
+
+  useEffect(() => {
+    const getProviders = async () => {
+      const result = await atmosphere.fetchQuery<useIntegrationProvidersGetProvidersListQuery>(
```

Comment:
-1 fetching data in `useEffect` isn't something we should be doing when using relay. this hook should have it's own fragment, and the fragment should be spread in each component that's using this hook

Reply:
Hi @BartoszJarocki, wasn't able to spread a fragment specifically for `teamMember.integrations`  in `OutcomeCard.tsx` because doing so would require making extensive changes throughout other `fragments` in the same hierarchy just to include the  `teamId` argument. As a result, we have instead created a new `root` for the  providers list `query`  in the `IntegrationsBanner`